etcdserver/api/membership/errors.go

```diff
@@ -21,10 +21,11 @@ import (
 )
 
 var (
-	ErrIDRemoved     = errors.New("membership: ID removed")
-	ErrIDExists      = errors.New("membership: ID exists")
-	ErrIDNotFound    = errors.New("membership: ID not found")
-	ErrPeerURLexists = errors.New("membership: peerURL exists")
+	ErrIDRemoved       = errors.New("membership: ID removed")
+	ErrIDExists        = errors.New("membership: ID exists")
+	ErrIDNotFound      = errors.New("membership: ID not found")
+	ErrPeerURLexists   = errors.New("membership: peerURL exists")
+	ErrPromotionFailed = errors.New("membership: promotion failed")
```

Comment:
Where is this error used? A promotion can fail if A) member ID not exist, B) member is already a voting member. Maybe what needed here is a more specific error when B) happens?

Reply:
This error is from joe's advise。 Add C） learner's progess doesn't catch up the other peer, refuse to promote now

Reply:
added